frankenstrings/frankenstrings.py

```diff
@@ -29,36 +29,43 @@
 Tags = dict[str, set[str]]
 B64Result = dict[str, tuple[int, bytes, bytes, bytes]]
 
+PDF_HEADER = b"%PDF-"
+
 # PE Strings
 PAT_EXEDOS = rb"(?s)This program cannot be run in DOS mode"
 PAT_EXEHEADER = rb"(?s)MZ.{32,1024}PE\000\000.+"
 
 BASE64_RE = rb"(?:[A-Za-z0-9+/]{10,}(?:&#(?:xA|10);)?[\r]?[\n]?){2,}[A-Za-z0-9+/]{2,}={0,2}"
 
 
+def extract_pdf_content(file_contents: bytes) -> bytes:
+    """Extract a PDF file from a file that mixes PDF with other data .
+
+    PDF file parsing is notoriously loose, allowing data that is not part of the PDF to be interspersed.
+    This function extracts the PDF part of a file that mixes PDF content with other data.
+
+    Currently just removes everything before the PDF Header.
+    """
+    offset = file_contents.find(PDF_HEADER, 0, 1023 + len(PDF_HEADER))  # Acrobat looks for the header in the first 1024 bytes
+    if offset < 1:  # Header is either missing or at the start like a normal pdf file
+        return b""
+    pdf_file = file_contents[offset:]
+    mime = magic.Magic(mime=True)
```

Comment:
I'm wondering if you need to recreate the mime object every time. Could you load it once, and reuse it?

Reply:
Actually checking locally it seems like libmagic just looks for the `%PDF-` and ignores everything else making it redundant.

Reply:
Also looking at the [Usage Notes](https://github.com/ahupp/python-magic?tab=readme-ov-file#usage) it seems like using the magic class at all is unecessary unless custom configuration is required.